src/database/models/team_item.rs

```diff
@@ -100,11 +108,230 @@ impl TeamMember {
+        .try_filter_map(|e| async {
+            Ok(e.right().map(|m| TeamMember {
+                id: TeamMemberId(m.id),
+                team_id: TeamId(m.team_id),
+                user_id: id,
+                name: m.member_name,
+                role: m.role,
+                permissions: Permissions::from_bits_truncate(m.permissions as u64),
+                accepted: m.accepted,
+            }))
+        })
+        .try_collect::<Vec<TeamMember>>()
+        .await?;
+
+        Ok(team_members)
+    }
+
+    pub async fn get_from_user_id<'a, 'b, E>(
+        id: TeamId,
+        user_id: UserId,
+        executor: E,
+    ) -> Result<Option<Self>, super::DatabaseError>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres>,
+    {
+        let result = sqlx::query!(
+            "
+            SELECT id, user_id, member_name, role, permissions, accepted
+            FROM team_members
+            WHERE (team_id = $1 AND user_id = $2 AND accepted = TRUE)
+            ",
+            id as TeamId,
+            user_id as UserId
+        )
+        .fetch_optional(executor)
+        .await?;
+
+        if let Some(m) = result {
+            Ok(Some(TeamMember {
+                id: TeamMemberId(m.id),
+                team_id: id,
+                user_id,
+                name: m.member_name,
+                role: m.role,
+                permissions: Permissions::from_bits_truncate(m.permissions as u64),
+                accepted: m.accepted,
+            }))
+        } else {
+            Ok(None)
+        }
+    }
+
+    pub async fn insert(
+        &self,
+        transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+    ) -> Result<(), sqlx::error::Error> {
+        sqlx::query!(
+            "
+            INSERT INTO team_members (
+                id, user_id, member_name, role, permissions, accepted
+            )
+            VALUES (
+                $1, $2, $3, $4, $5,
+                $6
+            )
+            ",
+            self.id as TeamMemberId,
+            self.user_id as UserId,
+            self.name,
+            self.role,
+            self.permissions.bits() as i64,
+            self.accepted
+        )
+        .execute(&mut *transaction)
+        .await?;
+
+        Ok(())
+    }
+
+    pub async fn delete<'a, 'b, E>(
+        id: TeamId,
+        user_id: UserId,
+        executor: E,
+    ) -> Result<(), super::DatabaseError>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres>,
+    {
+        sqlx::query!(
+            "
+            DELETE FROM team_members
+            WHERE (team_id = $1 AND user_id = $2 AND NOT role = $3)
+            ",
+            id as TeamId,
+            user_id as UserId,
+            crate::models::teams::OWNER_ROLE
+        )
+        .execute(executor)
+        .await?;
+
+        Ok(())
+    }
+
+    pub async fn edit_team_member<'a, 'b, E>(
+        id: TeamId,
+        user_id: UserId,
+        new_permissions: Option<i64>,
```

Comment:
This should take the permissions bitset rather than just an integer.  Using an integer here allows us to potentially insert invalid bit patterns into the database.